app/controllers/admin/vouchers_controller.rb

```diff
@@ -9,19 +9,33 @@ def new
     end
 
     def create
-      @voucher = Voucher.create(permitted_resource_params.merge(enterprise: @enterprise))
+      case params[:voucher][:voucher_type]
+      when "Vouchers::FlatRate"
+        @voucher =
+          Vouchers::FlatRate.create(permitted_resource_params.merge(enterprise: @enterprise))
+      when "Vouchers::PercentageRate"
+        @voucher =
+          Vouchers::PercentageRate.create(permitted_resource_params.merge(enterprise: @enterprise))
+      else
+        @voucher.errors.add(:type)
+        return render_error
+      end
```

Comment:
You could `constantize` the type string after checking it against the known types. Maybe worth if we add more types.

Reply:
something like ?  
```ruby
voucher_types = ["Vouchers::FlatRate", "Vouchers::PercentageRate"]
if voucher_types.include?(params[:voucher][:voucher_type])
  @voucher = params[:voucher][:voucher_type].constantize.create(permitted_resource_params.merge(enterprise: @enterprise))
else 
  @voucher.errors.add(:type)
  return render_error
end
```

Reply:
@mkllnk I think this is just waiting on your re-review of the fix for this comment (see newer commit https://github.com/openfoodfoundation/openfoodnetwork/pull/10821/commits/229ae91f3f947e825386d1d6063bbcb73ebbd700)